_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,217 @@
+
+## Clone and configure Kafka Connector
+
+Pull the `kafka-connector` code from github:
+
+```bash
+$ git clone https://github.com/openfaas-incubator/kafka-connector.git
+```
+
+Inside `kafka-connector` folder navigate to `yaml/kubernetes`:
+
+```bash
+$ cd kafka-connector/yaml/kubernetes
+```
+
+The file `connector-dep.yaml` contains all the configuration you need to set for your connector.
+
+It looks something like:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: kafka
+    component: kafka-connector
+  name: kafka-connector
+  namespace: openfaas
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kafka
+        component: kafka-connector
+    spec:
+      containers:
+      - name: kafka
+        image: openfaas/kafka-connector:0.4.0
+        env:
+          - name: gateway_url
+            value: "http://gateway.openfaas:8080"
+          - name: topics
+            value: "faas-request,"
+          - name: print_response
+            value: "true"
+          - name: print_response_body
+            value: "true"
+          - name: basic_auth
+            value: "true"
+          - name: secret_mount_path
+            value: "/var/secrets/"
+          - name: kafka_broker
+            value: "kafka"
+        volumeMounts:
+            - name: auth
+              readOnly: true
+              mountPath: "/var/secrets/"
+      volumes:
+      - name: auth
+        secret:
+          secretName: basic-auth
+```
+
+If you deployed your OpenFaaS instance with Helm and default values there is no need to make changes to this file. The `kafka_broker` variable is our broker host from which we will pull topics.
+
+In our case the broker is located within a service called `kafka`, which resides in the same namespace as the OpenFaaS gateway and uses the default kafka port of `9092`.
+
+The environmental variable `topics` is the place where we list the topics in which we are interested to receive from the broker. We will be using single topic for our example, but feel free to use as much as you wish, just separate them with comma like so `topics=example-topic-1,example-topic-2`.
+
+## Deploy the components
+
+In order to deploy the whole stack and play around with it go ahead and apply the kubernetes files while inside the kafka-connector folder:
+
+```bash
+$ kubectl apply -f yaml/kubernetes
+```
+The Broker deployed from the stack above contains a Producer in itself, but it is nested inside folders in the pods, so instead of searching it we will deploy simple producer provided by Alex:
+
+```bash
+$ kubectl run -n openfaas -t -i --image alexellis2/kafka-producer:0.2.0 producer
+```
+
+Congratulations now you have the whole stack deployed:
+* Zookeeper
+* Producer Service
+* Broker Host
+* Kafka Connector
+
+Before we proceed, make sure you have all the components running:
+
+```bash
+$ kubectl get pods -n openfaas
```

Comment:
Please make this just look for the specific parts related to the post.

Reply:
Done 👍 
```
kubectl get pods -n openfaas | grep -E 'kafka|zookeeper|producer'
kafka-broker-544fbccd48-p6pgr     1/1     Running            3          71m
kafka-connector-5d9f447f5-5drv7   1/1     Running            0          69m
zookeeper-699f568f6f-6b4n2        1/1     Running            0          71m
```